sklearn/feature_selection/_rfe.py

```diff
@@ -185,8 +190,12 @@ def _fit(self, X, y, step_score=None):
         n_features = X.shape[1]
         if self.n_features_to_select is None:
             n_features_to_select = n_features // 2
-        else:
+        elif self.n_features_to_select < 0:
+            raise ValueError("n_features_to_select must be positive or None")
+        elif isinstance(self.n_features_to_select, numbers.Integral):
             n_features_to_select = self.n_features_to_select
+        else:  # float
+            n_features_to_select = int(n_features * self.n_features_to_select)
```

Comment:
At this point, it would be good to check for if the float is > 1.0 and error in those cases.